internal/service/apigateway/authorizer_data_source.go

```diff
@@ -0,0 +1,73 @@
+package apigateway
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func DataSourceAuthorizer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAuthorizerRead,
+		Schema: map[string]*schema.Schema{
+			"rest_api_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"auth_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authorizer_uri": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"identity_source": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authorizer_result_ttl_in_seconds": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAuthorizerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).APIGatewayConn
+
+	restApiId := d.Get("rest_api_id").(string)
+	name := d.Get("name").(string)
+
+	authorizers, err := conn.GetAuthorizers(&apigateway.GetAuthorizersInput{
```

Comment:
what if there is more than one authorizer per api?

Reply:
i think this data source should be a plural data source for getting all authorizers for a given api

Reply:
there can be - but the name is a unique constraint. This is why I added the name filter.

Reply:
will pluralise

Reply:
done

Reply:
@DrFaust92 - is there anything else waiting on this PR?